packages/linux-firmware/linux-firmware.spec

```diff
@@ -53,10 +49,471 @@ mkdir -p %{buildroot}/%{fwdir}/updates
 
 # Use zstd compression for firmware files to reduce size on disk. This relies on
 # kernel support through FW_LOADER_COMPRESS (and FW_LOADER_COMPRESS_ZSTD for kernels >=5.19)
-make DESTDIR=%{buildroot}/ FIRMWAREDIR=%{fwdir} install-zst
+install -d %{buildroot}/%{fwdir}
+./copy-firmware.sh --zstd --ignore-duplicates %{buildroot}/%{fwdir}
 
 %files
 %dir %{fwdir}
-%{fwdir}/*
-%license LICENCE.* LICENSE.* GPL* WHENCE
+%dir %{fwdir}/updates
+
+# Directory structures for firmware we DO want to include.
+# These directories are intentionally NOT excluded so we can include specific files from them.
+%dir %{fwdir}/bnx2x
+%dir %{fwdir}/cxgb4
+%dir %{fwdir}/cxgb4/configs
+%dir %{fwdir}/i915
+%dir %{fwdir}/intel
+%dir %{fwdir}/intel/ice
+%dir %{fwdir}/intel/ice/ddp
+%dir %{fwdir}/intel/ice/ddp-comms
+%dir %{fwdir}/intel/ice/ddp-wireless_edge
+%dir %{fwdir}/qed
+%dir %{fwdir}/tigon
+
+# ALLOWLIST APPROACH: Exclude unwanted firmware, then explicitly include what we need
+#
+# We use an allowlist (default-deny) approach for security and maintainability:
+# - New firmware added upstream is automatically excluded (safe default)
+# - We only ship exactly what we need
+# - Package size stays minimal and controlled
+#
+# Why we can't use a single "%exclude %{fwdir}/*":
+# - That would exclude all subdirectories (bnx2x/, i915/, cxgb4/, etc.)
+# - Once a directory is excluded, you cannot include files from it
+# - Instead, we exclude specific unwanted directories and file patterns
+# - This preserves the directory structure we need (listed above)
+# - Then we explicitly include only the specific files we want
+#
+# Structure of excludes below:
+# 1. File pattern excludes (*.fw.zst, *.hex.zst, etc.) - catches loose files
+# 2. Directory excludes - removes entire directories we don't need
+# 3. Specific file pattern excludes - targeted removals for wireless, misc, etc.
```

Comment:
I would delete these long comments; they won't hold up well to refactoring, since you can't enforce structure through comments.